packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/app/{{project_name}}.dart

```diff
@@ -48,15 +50,57 @@ __MyMaterialAppState createState() => __MyMaterialAppState();
 }
 
 class __MyMaterialAppState extends State<_MyMaterialApp> {
+  Locale? _locale;
 
   @override
-  void initState() { {{#push_notifications}}
+  void initState() {
+    {{^enable_change_language}}
+    _locale = const Locale('en'); {{/enable_change_language}}
+    {{#enable_change_language}}
+    _setInitialLocale();
+    _updateLocale(); {{/enable_change_language}} {{#push_notifications}}
     _configureFCM(); {{/push_notifications}}
     _configureInterceptors();
-
     super.initState();
-  }{{#push_notifications}}
+  }
+
+  {{#enable_change_language}}
+  Future<void> _setInitialLocale() async {
+    final current = await context
+        .read<SharedPreferencesInstance>()
+        .getString('languageCurrent');
+    if (current != null) {
+      if (current.contains('EN')) {
+        setState(() {
+          _locale = const Locale('en');
+        });
+      } else {
+        setState(() {
+          _locale = const Locale('bg');
+        });
+      }
+    }
+    if(current == null){
+      setState(() {
+        _locale = const Locale('en');
+      });
+    }
+  } {{/enable_change_language}}
+
+  {{#enable_change_language}}
+  void _updateLocale() {
+  context
```

Comment:
Is there any reason to cause a memory leak here ?

Reply:
I have assigned the stream to a StreamSubscription<LanguageModel> _languageSubscription; and I call the   cancel() method on it in the dispose() method